include/deal.II/hp/fe_collection.h

```diff
@@ -1006,13 +1006,11 @@ namespace hp
   FECollection<dim, spacedim>::hp_constraints_are_implemented() const
   {
     Assert(finite_elements.size() > 0, ExcNoFiniteElements());
-
-    bool hp_constraints = true;
-    for (unsigned int i = 0; i < finite_elements.size(); ++i)
-      hp_constraints =
-        hp_constraints && finite_elements[i]->hp_constraints_are_implemented();
-
-    return hp_constraints;
+    return std::all_of(finite_elements.cbegin(),
```

Comment:
nice job -- this also leads to the early exit if one FE fails the test, which the programmer of that code missed!